cmd/lp-colouring/colouring.go

```diff
@@ -0,0 +1,169 @@
+package main
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"sync/atomic"
+
+	"github.com/ScottSallinen/lollipop/graph"
+	"github.com/kelindar/bitmap"
+)
+
+const EmptyColour = math.MaxUint32
+
+type VertexProperty struct {
+	NbrColours sync.Map
+	WaitCount  int64
+	Colour     uint32
+}
+
+type EdgeProperty struct{}
+
+func (p *VertexProperty) String() string {
+	s := fmt.Sprintf("{%d,%d,[", p.Colour, p.WaitCount)
+	p.NbrColours.Range(func(key, value any) bool {
+		s += fmt.Sprintf("%d:%d,", key, value)
```

Comment:
We probably want to standardize all string outputs, e.g. to conform to CSV for easy parsing / importing for data analysis.